tools/generate_chain.py

```diff
@@ -0,0 +1,142 @@
+import cProfile
+import random
+import sqlite3
+import time
+from contextlib import closing, contextmanager
+from pathlib import Path
+from typing import Iterator, List
+
+import zstd
+
+from chia.types.blockchain_format.coin import Coin
+from chia.types.spend_bundle import SpendBundle
+from chia.util.chia_logging import initialize_logging
+from chia.util.ints import uint64
+from chia.util.path import mkdir
+from tests.block_tools import create_block_tools
+from tests.util.keyring import TempKeyring
+from tools.test_constants import test_constants
+
+
+@contextmanager
+def enable_profiler(profile: bool) -> Iterator[None]:
+    if not profile:
+        yield
+        return
+
+    with cProfile.Profile() as pr:
+        yield
+
+    pr.create_stats()
+    pr.dump_stats("generate-chain.profile")
+
+
+root_path = Path("./test-chain").resolve()
+mkdir(root_path)
+with TempKeyring() as keychain:
+
+    bt = create_block_tools(constants=test_constants, root_path=root_path, keychain=keychain)
+    initialize_logging(
+        "generate_chain", {"log_level": "DEBUG", "log_stdout": False, "log_syslog": False}, root_path=root_path
+    )
+
+    with closing(sqlite3.connect("stress-test-blockchain.sqlite")) as db:
+
+        print("initializing v2 block store")
+        db.execute(
+            "CREATE TABLE full_blocks("
+            "header_hash blob PRIMARY KEY,"
+            "prev_hash blob,"
+            "height bigint,"
+            "in_main_chain tinyint,"
+            "block blob)"
+        )
+
+        wallet = bt.get_farmer_wallet_tool()
+        coinbase_puzzlehash = wallet.get_new_puzzlehash()
+
+        blocks = bt.get_consecutive_blocks(
+            3,
+            farmer_reward_puzzle_hash=coinbase_puzzlehash,
+            pool_reward_puzzle_hash=coinbase_puzzlehash,
+            guarantee_transaction_block=True,
+            genesis_timestamp=uint64(1234567890),
+            time_per_block=30,
+        )
+
+        unspent_coins: List[Coin] = []
+
+        for b in blocks:
+            for coin in b.get_included_reward_coins():
+                if coin.puzzle_hash == coinbase_puzzlehash:
+                    unspent_coins.append(coin)
+            db.execute(
```

Comment:
it'd be faster to use `executemany` here

Reply:
probably, but it wouldn't affect the runtime of the program (because the disk write is negligible), and it was more important to me to have the block written to disk as soon as possible after it was done, so I could run a test against it. to ensure the blocks validate successfully.